src/main/java/seedu/duke/command/DeleteCommand.java

```diff
@@ -58,24 +46,27 @@ public String extractFolderName(String input) {
      */
     @Override
     public void execute(SecretMaster secureNUSData) throws SecretNotFoundException {
-        Secret deleteData = null;
-        boolean isValid = false;
-        try {
-            deleteData = secureNUSData.getByName(secretName);
-            isValid = true;
-        } catch (SecretNotFoundException e) {
-            Ui.printError("Data not found!");
-            isValid = false;
-        }
-        if (isValid && (deleteData != null)) {
-            System.out.println("You deleted " + secretName + " in folder: " + folderName);
+        for (int index = 1; index < secretNames.length; index += 1) {
+            secretName = secretNames[index];
+            Secret deleteData = null;
+            boolean isValid = false;
             try {
-                secureNUSData.removeSecret(deleteData);
+                deleteData = secureNUSData.getByName(secretName);
+                isValid = true;
             } catch (SecretNotFoundException e) {
                 Ui.printError("Data not found!");
+                isValid = false;
+            }
+            if (isValid && (deleteData != null)) {
```

Comment:
This nesting is quite problematic. So many things with over each other. There are also some Exceptions arising that falls through